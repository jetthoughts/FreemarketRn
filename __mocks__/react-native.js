const rn = require('react-native');

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.fn(),
    canOpenURL: jest.fn(),
    getInitialURL: jest.fn(),
  };
});

jest.mock('../app/api/firebase', () => {
  return {
    saveImageToStorage: jest.fn(),
  }
});

const helpers = require('../app/lib/helpers');
helpers.dateToTimeString = () => '10:20';

module.exports = rn;
