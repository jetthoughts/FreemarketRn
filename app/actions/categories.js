export const loadCategories = () => ({ type: 'LOAD_CATEGORIES' });
export const selectCategory = (id) => ({ type: 'SELECT_CATEGORY', id });
