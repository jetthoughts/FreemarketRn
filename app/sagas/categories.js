import { put, call, takeLatest } from 'redux-saga/effects';

import {
  categoriesSuccess,
  categoriesError,
  requestCategories,
} from '../reducers/categories';
import { fetchAllCategories } from '../Api';

export function* fetchCategories() {
  try {
    const response = yield call(fetchAllCategories);
    const [, ...categories] = response.data;
    yield put(categoriesSuccess(categories));
  } catch (e) {
    yield put(categoriesError(e));
  }
}

export function* watchCategoriesRequest() {
  yield takeLatest(requestCategories.getType(), fetchCategories);
}
