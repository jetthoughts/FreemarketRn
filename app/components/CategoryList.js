/* eslint react/prop-types: 0 */

import React, { Component } from 'react';
import { ListView } from 'react-native';

import CategoryListRow from './CategoryListRow';
import categories from '../constants/categories';

export default class CategoryList extends Component {
  constructor(props) {
    super(props);

    const ds = new ListView.DataSource({
      rowHasChanged: (r1, r2) => r1 !== r2,
    });
    this.dataSource = ds.cloneWithRows(categories);
  }

  render() {
    const renderRow = (rowData, _, rowID) => {
      const { onPressCategory } = this.props;
      return <CategoryListRow {...{ rowData, rowID, onPressCategory }} />;
    };

    return <ListView dataSource={this.dataSource} renderRow={renderRow} />;
  }
}
