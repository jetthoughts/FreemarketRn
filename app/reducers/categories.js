import { combineReducers } from 'redux';
import { createAction, createReducer } from 'redux-act';
import categories from '../constants/categories';

export const loadCategories = createAction('load categories list');
export const selectCategory = createAction('change current category');

const listReducer = createReducer({
  [loadCategories]: (state) => categories,
}, []);

const currentReducer = createReducer({
  [selectCategory]: (state, payload) => payload,
}, 0)

export default combineReducers({
  list: listReducer,
  current: currentReducer,
});
