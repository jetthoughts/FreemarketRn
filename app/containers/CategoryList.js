import { connect } from 'react-redux';

import CategoryListView from '../components/CategoryListView';
import { requestCategories, selectCategory } from '../reducers/categories';

const mapStateToProps = state => ({
  categories: state.categories.list.records,
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  loadCategories: () => dispatch(requestCategories()),
  onCategoryPress: (category) => {
    dispatch(selectCategory(category));
    ownProps.afterCategoryPress();
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(CategoryListView);
