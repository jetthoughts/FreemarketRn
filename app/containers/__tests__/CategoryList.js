import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import CategoryList from '../CategoryList';
import categories from '../../constants/categories';

it('renders successfully', () => {
  const tree = renderer.create(
    <CategoryList categories={categories} onPressCategory={() => true} />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
