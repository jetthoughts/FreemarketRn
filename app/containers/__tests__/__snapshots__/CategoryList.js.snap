// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders successfully 1`] = `
<RCTScrollView
  dataSource={
    ListViewDataSource {
      "items": 5,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}
>
  <View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Antiques
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Books, Comics & Magazines
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Cars, Motorcycles & Vehicles
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Events Tickets
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Pet Supplies
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
